school_hostel/models/hostel.py

```diff
@@ -0,0 +1,540 @@
+    ):
+        """Override method to get hostel of student selected"""
+        if self._context.get("student_id"):
+            stud_obj = self.env["student.student"]
+            stud = stud_obj.browse(self._context["student_id"])
+            datas = []
+            if stud.gender:
+                self._cr.execute(
+                    """select id from hostel_type where type=%s or
+                type='common' """,
+                    (stud.gender,),
+                )
+                data = self._cr.fetchall()
+                for d in data:
+                    datas.append(d[0])
+            args.append(("id", "in", datas))
+        return super(HostelType, self)._search(
+            args=args,
+            offset=offset,
+            limit=limit,
+            order=order,
+            count=count,
+            access_rights_uid=access_rights_uid,
+        )
+
+
+class HostelRoom(models.Model):
+
+    _name = "hostel.room"
+    _description = "Hostel Room Information"
+    _rec_name = "room_no"
+
+    @api.depends("student_per_room", "student_ids")
+    def _compute_check_availability(self):
+        """Method to check room availability"""
+        for rec in self:
+            rec.availability = rec.student_per_room - len(rec.student_ids.ids)
+
+    name = fields.Many2one("hostel.type", "HOSTEL", help="Name of hostel")
+    floor_no = fields.Integer("Floor No.", default=1, help="Floor Number")
+    room_no = fields.Char("Room No.", required=True)
+    student_per_room = fields.Integer(
+        "Student Per Room", required=True, help="Students allocated per room"
+    )
+    availability = fields.Float(
+        compute="_compute_check_availability",
+        store=True,
+        string="Availability",
+        help="Room availability in hostel",
+    )
+    rent_amount = fields.Float(
+        "Rent Amount Per Month", help="Enter rent amount per month"
+    )
+    hostel_amenities_ids = fields.Many2many(
+        "hostel.amenities",
+        "hostel_room_amenities_rel",
+        "account_id",
+        "tax_id",
+        string="Hostel Amenities",
+        domain="[('active', '=', True)]",
+        help="Select hostel roon amenities",
+    )
+    student_ids = fields.One2many(
+        "hostel.student", "room_id", string="Students", help="Enter students"
+    )
+
+    _sql_constraints = [
+        ("room_no_unique", "unique(room_no)", "Room number must be unique!"),
+        (
+            "floor_per_hostel",
+            "check(floor_no < 10)",
+            "Error ! Floor per HOSTEL should be less than 10.",
+        ),
+        (
+            "student_per_room_greater",
+            "check(student_per_room < 10)",
+            "Error ! Student per room should be less than 10.",
+        ),
+    ]
+
+    @api.model
+    def fields_view_get(
+        self, view_id=None, view_type="form", toolbar=False, submenu=False
+    ):
+        """Inherited this method to make edit button invisible in list view"""
```

Comment:
Do we need this?
Cannot we simply block this by right access control ?
MODEL with this group has only READ access.
@AnuKPatel

Reply:
Yes, This method might defined separately because, If the user with this group will also have the access of school admin then all the access would be overtaken by that for this model.
But yes, as school admin should have all the access, this method can be discarded.